waiter/container_delete.go

```diff
@@ -0,0 +1,73 @@
+package waiter
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/nspcc-dev/neofs-sdk-go/client"
+	apistatus "github.com/nspcc-dev/neofs-sdk-go/client/status"
+	"github.com/nspcc-dev/neofs-sdk-go/container"
+	cid "github.com/nspcc-dev/neofs-sdk-go/container/id"
+)
+
+// ContainerDeleteExecutor represents requirements to async container delete operation.
+type ContainerDeleteExecutor interface {
+	ContainerDelete(ctx context.Context, id cid.ID, prm client.PrmContainerDelete) error
```

Comment:
Good question. There is possible many copy-paste documentation. May be we may use some `implements` words in function comments or `// See documentation for function in [client.Client.ContainerDelete].`

Reply:
I'd say that commenting every method is not really required here, we can refer to `client.Client` in `ContainerDeleteExecutor` description and say that the same semantics is expected. If this would be some _external_ (not implemented in SDK) interface we're expecting, that'd be a different story.

Reply:
Left reference to client.Client for each executor